server/autobuild

```diff
@@ -364,6 +364,22 @@ os_msys2() {
 os_macos() {
     if ! which uname >/dev/null 2>&1 || [ "$(uname -s)" != "Darwin" ]; then
         return 1
+    elif which nix-env >/dev/null 2>&1; then
+        # Already in the nix-shell.
+        if [ -n "$AUTOBUILD_NIX_SHELL" ]; then
+            return 0
+        fi
+        if ! which nix-shell >/dev/null 2>&1; then
+            return 1
+        fi
+        if [ -n "$DRY_RUN" ]; then
+            return 0
+        fi
+        command="AUTOBUILD_NIX_SHELL=true"
+        command="$command;export AUTOBUILD_NIX_SHELL"
+        command="$command;$(quote "$0" "$@")"
+        exec nix-shell --pure --command "$command" \
+           -p pkg-config popplar automake
```

Comment:
please create  PKGCMD, PKGARGS, PACKAGES variables if possible (or at least PACKAGES in this case). Also, there is a spelling mistake here for poppler package. Please make sure that these args are correct as the CI will not protect against these errors. 

If you can extend the CI rules to run on nixos and / or on mac with nix installed, that will be a huge bonus point for me. (can be in a separate PR)